### PR TITLE
Template improvements

### PR DESCRIPTION
**Pull request by andradefil**
# Big Changes
Improve layout css and adds more description about the budget features
Screen Shot 2021-08-21 at 14 03 02

`templates` folder separated by country `br`, `us` etc.. today have only br

# Small Changes
* Refactor `index.js` to have only the call to `app.js` function and parse args
* Adds centralized log file to have the rules of log centralized in one place
* Adds function `readFile` to be possible read html templates from file in disk

### CODE REVIEW

File: src/file.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    }  catch (e) {
+        console.log(e);
+        throw e
+    }
+}
+
+function writeFile(filePath, fileName, fileContent) {
+    try {
+        if (!fs.existsSync(filePath)) {
+            console.warn("Output dir doesnt exists. Creating it...");
```
Comment (Thrashattack) [inline src/file.js:15]: I Belive we can use the new log module here so we can centralize the execution of console.log and console.warn

File: src/file.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            fs.mkdirSync(filePath, '0777');
+        }
+        fs.writeFileSync(filePath + fileName, fileContent);
+        return true;
+
+    } catch (e) {
+        console.log(e);
```
Comment (Thrashattack) [inline src/file.js:22]: Here aswell log.error

File: src/app.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    const content = generator.template({
+        customerName: config.customerName,
+        features: config.features,
+        totalTime: budget.totalTime,
+        taxAmount: budget.taxAmount,
+        amount: budget.amount,
+        cost: budget.cost,
+    });
+    log.verbose("Budget created, generating file...");
+    const success = writeFile(args.outputDir, "budget.html", content);
```
Comment (Thrashattack) [inline src/app.js:31]: what do you think about `const OUTPUT_FILENAME = "budget.html"` ?

File: src/generator.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+const HandleBars = require("handlebars");
+var path = require('path');
+var appDir = path.dirname(require.main.filename);
+const { readFile } = require('./file');
+
+function template(payload) {
+    const file = readFile(appDir + '/src/templates/br.html')
```
Comment (Thrashattack) [inline src/generator.js:7]: Maybe this `br` can be passed to payload as `locale` prop comming right from the args as `-l` or `--locale` and have the default setted to `br`.

**Review by Thrashattack — COMMENTED** [review]
Great work Filipinas, now we can think in expand in a greater number of paths. 
I just feel that we could set up some locale args by now and use the new log module as the only way to output to console.

**Comment by camilomelges** [comment]
parser.js show error when you run application with ```node index.js```

```
Couldn't load the config file
Error: Cannot find module './allspark-config.json'
```

**Comment by Thrashattack** [comment]
> parser.js show error when you run application with `node index.js`
> 
> ```
> Couldn't load the config file
> Error: Cannot find module './allspark-config.json'
> ```

You need to set up a json file called allspark-config.json in the project root or give it a path to another config file with the flag `-c` or `--configPath /path/to/my/config.json` 
The file specification is described at [Readme](https://github.com/filipewvandrade/allspark/blob/main/README.md)

**Comment by camilomelges** [comment]
Yep, but is interesting you show a more descriptive message in this case for example, and remove "config.json" from src too, but this is not a obrigatory, message is.

**Pull request merged**